### ISSUE TITLE
[Data] Curate African voice training dataset

### ISSUE THREAD

**Issue by wallscaler** [wallscaler/baobab-tts#3]
# Issue #3: [Data] Curate African Voice Training Dataset

## Labels
`data`, `priority-critical`, `phase-1`

## Objective

Curate and prepare a high-quality training dataset from African speech corpora, implementing the quality-driven selection approach for optimal training efficiency.

## Background

We have access to several African speech datasets:
- **NaijaVoices** (1800h) - Nigerian languages
- **BibleTTS** (86h) - 10 African languages, studio quality
- **1000 African Voices** (100h) - Multi-speaker

Following the [5% rule](https://arxiv.org/abs/2407.00102), we'll select the highest-quality samples rather than using everything.

## Data Sources

| Dataset | Languages | Hours | Priority |
|---------|-----------|-------|----------|
| NaijaVoices | Yoruba, Hausa, Igbo | 1,800 | P0 |
| BibleTTS | 10 African languages | 86 | P0 |
| 1000 African Voices | 12 languages | 100 | P1 |
| LibriTTS (baseline) | English | 300 | P1 |
| Common Voice | Swahili, Kinyarwanda | 500 | P2 |

### 1. Download and Verify

```bash
# Download all datasets
baobab data download naijavoices
baobab data download bibletss
baobab data download african-voices
baobab data download libritts --subset clean-360
```

### 2. Quality Scoring

```python
from baobab.data import QualityScorer

scorer = QualityScorer(
    metrics=[
        "snr",                # Signal-to-noise ratio
        "mos_predictor",      # Predicted mean opinion score
        "speaker_consistency",# Within-speaker similarity
        "text_alignment",     # Audio-text alignment score
    ]
)

# Score all samples
for sample in dataset:
    sample.quality_score = scorer.score(sample)
```

### 3. Language Balancing

Target distribution for Pan-African coverage:

```python
target_distribution = {
    # Nigerian (primary - largest dataset)
    "yoruba": 0.20,
    "hausa": 0.15,
    "igbo": 0.15,

    # East African
    "swahili": 0.12,
    "amharic": 0.08,
    "kinyarwanda": 0.05,

    # Southern African
    "zulu": 0.05,
    "xhosa": 0.05,

    # Other West African
    "wolof": 0.05,
    "lingala": 0.05,

    # Baseline (stability)
    "english": 0.05,
}
```

### 4. Quality-Driven Selection

```python
# Select top 5% by quality within each language
for language in languages:
    lang_samples = dataset.filter(language=language)
    threshold = np.percentile(lang_samples.quality_scores, 95)
    selected = lang_samples.filter(quality >= threshold)
```

## Expected Output

| Language | Raw Hours | Selected Hours | Samples |
|----------|-----------|----------------|---------|
| Yoruba | 600 | 100 | ~15,000 |
| Hausa | 600 | 75 | ~12,000 |
| Igbo | 600 | 75 | ~12,000 |
| Swahili | 200 | 60 | ~10,000 |
| Amharic | 50 | 40 | ~6,000 |
| Others | 400 | 100 | ~15,000 |
| **Total** | **~2,500** | **~450** | **~70,000** |

## Files to Create

- [ ] `src/data/__init__.py`
- [ ] `src/data/loaders/naijavoices.py`
- [ ] `src/data/loaders/bibletss.py`
- [ ] `src/data/loaders/african_voices.py`
- [ ] `src/data/quality_scorer.py`
- [ ] `src/data/language_balancer.py`
- [ ] `configs/data/african_mix.yaml`
- [ ] `scripts/prepare_data.py`

## Acceptance Criteria

- [ ] All datasets download successfully
- [ ] Quality scoring pipeline works
- [ ] Language distribution matches target
- [ ] DataLoader integrates with training
- [ ] Documentation in `docs/datasets.md`

## Quality Thresholds

Minimum requirements for inclusion:
- SNR ≥ 20 dB
- Predicted MOS ≥ 3.5
- Audio duration: 1-30 seconds
- Text-audio alignment score ≥ 0.8

## References

- [NaijaVoices](https://arxiv.org/abs/2505.20564)
- [BibleTTS](https://arxiv.org/abs/2207.03546)
- [1000 African Voices](https://arxiv.org/abs/2406.11727)
- [Quality-Driven Selection](https://arxiv.org/abs/2407.00102)
